ksql-engine/src/main/java/io/confluent/ksql/function/udaf/window/WindowSelectMapper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.ksql.function.udaf.window;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.function.KsqlAggregateFunction;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.kafka.streams.kstream.ValueMapperWithKey;
+import org.apache.kafka.streams.kstream.Window;
+import org.apache.kafka.streams.kstream.Windowed;
+
+/**
+ * Used to handle the special cased {@link WindowStartTimeKudaf}
+ */
+public final class WindowSelectMapper
+    implements ValueMapperWithKey<Windowed<?>, GenericRow, GenericRow> {
+
+  private final Map<Integer, Type> windowSelects;
+
+  public WindowSelectMapper(
+      final Map<Integer, KsqlAggregateFunction> aggFunctionsByIndex) {
+    this.windowSelects = aggFunctionsByIndex.entrySet().stream()
+        .filter(e -> e.getValue().getFunctionName().equals(WindowStartTimeKudaf.getFunctionName()))
```

Comment:
This should be case insensitive comparison.

Reply:
done.